src/Messages/TSMessagesManager.m

```diff
@@ -633,6 +633,15 @@ - (TSIncomingMessage *)handleReceivedEnvelope:(OWSSignalServiceProtosEnvelope *)
       if (thread && incomingMessage) {
           [incomingMessage saveWithTransaction:transaction];
 
+          // Any messages sent from the current user - from this device or another - should be
+          // automatically marked as read.
+          if ([TSAccountManager isRegistered]) {
```

Comment:
I think this `isRegistered` is just a sanity check, but it's written in a way that makes it seem like it's an expected case (if anything we should be exploding in the omitted `else` because something is very, very, wrong at this point). For example, we could never download the message we're processing if we weren't registered.

My concern is that it unnecessarily leads the reader down a crazy path, and we'd be best to omit it altogether.